### PR TITLE
Added logging

### PR DESCRIPTION
**Pull request by GurliGebis**
This enables logging in the context menu handlers.
To enable logging, set the environment variable `NPPSHELL_LOGGING_ENABLED` to `true` and restart explorer.
To disable logging, set it to something else (or remove it) and restart explorer.

### CODE REVIEW

File: LoggingHelper.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+{
+    appDataLoggingEnabled = IsAppDataLoggingEnabled();
+
+    if (!appDataLoggingEnabled)
+    {
+        return;
+    }
+
+    const wstring logFileFolder = CreateAppDataFolder();
+    logFilePath = logFileFolder + L"\\NppShell.txt";
```
Comment (donho) [inline LoggingHelper.cpp:18]: Is it better to use **NppShell.log** instead of **NppShell.txt** ?

**Review by donho — CHANGES_REQUESTED** [review]
Please change the log file ext name from `txt` to `log`.

File: LoggingHelper.cpp (hunk 1 above)
Comment (GurliGebis) [reply LoggingHelper.cpp]: Yep, I have changed it 🙂

**Review by donho — APPROVED** [review]

**Comment by donho** [comment]
@GurliGebis 
FYI:
https://community.notepad-plus-plus.org/topic/24771/notepad-v8-5-5-release-candidate

**Pull request closed without merging**